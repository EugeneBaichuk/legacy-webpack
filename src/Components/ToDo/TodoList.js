import React, {memo} from 'react';
import { ListGroup, ListGroupItem, Button } from 'reactstrap';

const TodoList = ({questList, removeQuest, completed}) => (
  <ListGroup>
    {questList.map((quest, i) => (
      <ListGroupItem key={i}>
        {quest} &nbsp;
        <Button color="danger" size="sm" onClick={removeQuest(i)}>
          {completed}
        </Button>
      </ListGroupItem>
    ))}
  </ListGroup>
)

export default memo(TodoList);
