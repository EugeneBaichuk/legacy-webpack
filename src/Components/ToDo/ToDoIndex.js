import React, { useCallback, useState } from 'react';
import { InputGroup, Input, Button } from 'reactstrap';
import TodoList from './TodoList';

const QuestList = ({completed, addQuestText}) => {
  const [questList, setQuestList]  = useState([]);
  const [todoInputValue, setTodoInputValue] = useState('');

  const handleChangeTodoInputValue = (e) => {
    setTodoInputValue(e.target.value);
  }

  const addQuest = (e) => {
    e.preventDefault();
    setQuestList(list => [...list, todoInputValue ]);
    setTodoInputValue('');
  }

  const removeQuest = useCallback((questId) => () => {
    setQuestList((prevList) => prevList.filter((_, id) => id !== questId));
  },[setQuestList]);

  return (
    <div className="content">
      <TodoList 
        questList={questList}
        removeQuest={removeQuest} 
        completed={completed} 
      />
      <hr />
      <form className="form" id="addQuestForm" onSubmit={addQuest}>
        <InputGroup>
          <Input
            value={todoInputValue}
            onChange={handleChangeTodoInputValue}
            type="text"
            className="input"
            id="addInput"
            placeholder="add to-do list item"
          />
          <Button color="info" type="submit" disabled={!todoInputValue}>
            {addQuestText}
          </Button>
        </InputGroup>
      </form>
    </div>
  )
}

export default QuestList;