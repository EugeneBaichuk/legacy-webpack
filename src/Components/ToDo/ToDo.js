import React from 'react';
import Layout from '../Common/Layout';
import ToDoIndex from './ToDoIndex';
import { todoHeader, todoTasksList } from '../../Config/todo.constants';
import { completed, addQuestText } from '../../Config/todo.constants';

export const ToDo = () => (
  <Layout 
    header={todoHeader} 
    list={todoTasksList} 
  >
    <ToDoIndex completed={completed} addQuestText={addQuestText}/>
  </Layout>
)
