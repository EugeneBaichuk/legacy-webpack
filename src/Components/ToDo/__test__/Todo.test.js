import React from 'react';
import { render, cleanup } from '@testing-library/react';
import { ToDo } from '../ToDo';

describe('Cats component', () => {
  afterEach(cleanup);

  test('renders the header correctly', () => {
    const { asFragment } = render(<ToDo />);
    expect(asFragment(<ToDo />)).toMatchSnapshot();
  });
});