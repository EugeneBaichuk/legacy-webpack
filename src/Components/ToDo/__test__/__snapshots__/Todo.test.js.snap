// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cats component renders the header correctly 1`] = `
<DocumentFragment>
  <div
    class="layout container"
  >
    <h1
      class="mb-4 mt-4"
    >
      In this challenge your goal is to create a to do list:
    </h1>
    <ul
      class="list-group"
    >
      <li
        class="list-group-item"
      >
        ability to accept a task from a user
      </li>
      <li
        class="list-group-item"
      >
        ability to add this to the state
      </li>
      <li
        class="list-group-item"
      >
        ability to display all of the tasks
      </li>
      <li
        class="list-group-item"
      >
        ability to evaluated if a task was completed and update the state
      </li>
    </ul>
    <hr
      class="my-4"
    />
    <div
      class="content"
    >
      <ul
        class="list-group"
      />
      <hr />
      <form
        class="form"
        id="addQuestForm"
      >
        <div
          class="input-group"
        >
          <input
            class="input form-control"
            id="addInput"
            placeholder="add to-do list item"
            type="text"
            value=""
          />
          <button
            class="btn btn-info disabled"
            disabled=""
            type="submit"
          >
            Add Quest
          </button>
        </div>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
